pkg/objstore/s3/s3_test.go

```diff
@@ -3,25 +3,44 @@ package s3
 import (
 	"testing"
 	"time"
+
+	"github.com/improbable-eng/thanos/pkg/testutil"
 )
 
-func TestParseConfig(t *testing.T) {
-	inputConfig := []byte(`bucket: abcd
+func TestParseConfig_DefaultHTTPOpts(t *testing.T) {
+	input := []byte(`bucket: abcd
 insecure: false
 http_config:
   idle_conn_timeout: 50s`)
-	cfg, err := ParseConfig(inputConfig)
-	if err != nil {
-		t.Errorf("failed to parse config: %s", err)
-	}
+	cfg, err := parseConfig(input)
+	testutil.Ok(t, err)
+
 	if time.Duration(cfg.HTTPConfig.IdleConnTimeout) != time.Duration(50*time.Second) {
 		t.Errorf("parsing of idle_conn_timeout failed: got %v, expected %v",
 			time.Duration(cfg.HTTPConfig.IdleConnTimeout), time.Duration(50*time.Second))
 	}
+
 	if cfg.Bucket != "abcd" {
 		t.Errorf("parsing of bucket failed: got %v, expected %v", cfg.Bucket, "abcd")
 	}
 	if cfg.Insecure != false {
 		t.Errorf("parsing of insecure failed: got %v, expected %v", cfg.Insecure, false)
 	}
+
+}
+
+func TestValidate_OK(t *testing.T) {
```

Comment:
You many want to consider testing the failure case when when only one / none of `access_key` and `secret_key` are provided.

Reply:
true